src/entry.c

```diff
@@ -326,6 +326,11 @@ gboolean CacheEntryPasswordSet(CacheEntry *self, const gchar *password,
   return TRUE;
 }
 
+void CacheEntryHashSet(CacheEntry *self, CacheEntryAlgorithm algorithm,
+                  GBytes *hash, CacheEntryArgs *args) {
```

Comment:
Nothing is done with the args argument. Did you mean to copy them?

Make sure to consider ownership of the two GBytes *. That type is immutable, so you can either increase its reference count or take ownership so the caller is giving up its own reference. I recommend the first, and the documentation should make it clear. That code should be exercised in entry_test.c. 

Also, CacheEntryArgs is usually passed by value because it has no reference counting. I'd avoid making it a pointer here.